backend/public-graph/graph/resolver.go

```diff
@@ -2937,25 +2933,30 @@ func (r *Resolver) ProcessPayload(ctx context.Context, sessionSecureID string, e
 		(sessionHasErrors && (sessionObj.HasErrors == nil || !*sessionObj.HasErrors))
 
 	if doUpdate && !excluded {
-		fieldsToUpdate := model.Session{
-			PayloadUpdatedAt: &now, BeaconTime: beaconTime, HasUnloaded: hasSessionUnloaded, Processed: &model.F, ObjectStorageEnabled: &model.F, DirectDownloadEnabled: false, Chunked: &model.F, Excluded: false,
+		// By default, GORM will not update non-zero fields. This is undesirable for boolean columns.
+		// By explicitly specifying the columns to update, we can override the behavior.
+		// See https://gorm.io/docs/update.html#Updates-multiple-columns
+		if err := r.DB.Model(&model.Session{Model: model.Model{ID: sessionID}}).
+			Select("PayloadUpdatedAt", "BeaconTime", "HasUnloaded", "Processed", "ObjectStorageEnabled", "Chunked", "DirectDownloadEnabled", "Excluded", "ExcludedReason", "HasErrors").
+			Updates(&model.Session{
+				PayloadUpdatedAt:      &now,
+				BeaconTime:            beaconTime,
+				HasUnloaded:           hasSessionUnloaded,
+				Processed:             &model.F,
+				ObjectStorageEnabled:  &model.F,
+				DirectDownloadEnabled: false,
+				Chunked:               &model.F,
+				Excluded:              false,
+				ExcludedReason:        "",
+				HasErrors:             &sessionHasErrors,
+			}).Error; err != nil {
+			log.WithContext(ctx).Error(e.Wrap(err, "error updating session"))
+			return err
 		}
-
-		// We only want to update the `HasErrors` field if the session has errors.
```

Comment:
Did some clean up here. See the comment on the right as to why we have to specify the column names.